### PR TITLE
feature: Add `add_interpolation` assist

### PR DESCRIPTION
**Pull request by dotjpg3141**
This adds an assist, which adds interpolations in string literals.

### CODE REVIEW

**Review by Veykril — COMMENTED** [review]
I don't like saying this since I'm sure a lot of work has went into this but we already have this in form of postfix completions and I don't think I see value of having an assist here.

Code_IRTqGxwAFh

**Comment by flodiebold** [comment]
I think this can still be pretty useful since it also allows you to add interpolations to existing macro calls. (I basically never think to use the postfix completion, but this one I might remember to use.)

**Comment by Veykril** [comment]
Alright, though I am sure there is some potential for code re-use here.

File: crates/ide_assists/src/handlers/add_interpolation.rs [hunk 1]
```diff
@@ -0,0 +1,410 @@
...
+
+    for not in token_tree.token_trees_and_tokens() {
+        if let Some(token) = not.into_token() {
+            match (token.kind(), name_state) {
+                (syntax::SyntaxKind::IDENT, NameState::Identifier) => {
+                    name_candiate = Some(token);
+                    name_state = NameState::Equal;
+                }
+                (T![=], NameState::Equal) => {
+                    debug_assert!(name_candiate.is_some());
```
Comment (Veykril) [inline crates/ide_assists/src/handlers/add_interpolation.rs:165]: This debug_assert seems fishy, what if we have something like `format!("{foo}", =)`, not really a valid call but still something that could be written. So this assert should go I think

**Review by Veykril — COMMENTED** [review]
Sorry this took so long to get back to, I had a bunch of things on my plate the past weeks.

This seems mostly okay, its a bit too far on the AST-only front, but that's not really something we can change right now. We should also look into creating some general `format`-ish "parsing" tools as this is something we can probably re-use in other places.

With that said I think we can merge this with the `debug_assert` removed. Thanks!

**Comment by Veykril** [comment]
Closing this for now, if you find the time to come back to this feel free to re-open the issue.

**Pull request closed without merging**